### PR TITLE
Add docker build job in manifest e2e workflow

### PR DESCRIPTION
**Pull request by daisy-ycguo**
## Description

Add docker build job in manifest e2e workflow

## Issues

n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

n/a

## Tests

CI test

### CODE REVIEW

**Review by lvliang-intel — APPROVED** [review]

File: .github/workflows/scripts/build_push.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    fi
+    docker push ${IMAGE_REPO}opea/$1:$IMAGE_TAG
+    docker rmi ${IMAGE_REPO}opea/$1:$IMAGE_TAG
+}
+
+# $1 is like "apple orange pear"
+for MEGA_SVC in $1; do
+    case $MEGA_SVC in
+        "ChatQnA"|"CodeGen"|"CodeTrans"|"DocSum"|"SearchQnA")
+            cd $MEGA_SVC/docker
```
Comment (chensuyue) [inline .github/workflows/scripts/build_push.sh:50]: All the examples will be unified into the same code structure, so we can use the same build steps.

File: .github/workflows/manifest-e2e.yaml [hunk 2]
```diff
@@ -57,9 +57,19 @@ jobs:
           run_matrix=$run_matrix"]}"
           echo "run_matrix=${run_matrix}" >> $GITHUB_OUTPUT
 
-  manifest-test:
+  mega-image-build:
     needs: job1
     if: always() && ${{ needs.job1.outputs.run_matrix.include.length }} > 0
```
Comment (chensuyue) [inline .github/workflows/manifest-e2e.yaml:62]: If the matrix is empty, the test won't continue. So generally we don't need this line. e.g. https://github.com/opea-project/GenAIComps/actions/runs/9483916020/job/26132175597?pr=156

File: .github/workflows/scripts/build_push.sh (hunk 1 above)
Comment (daisy-ycguo) [reply .github/workflows/scripts/build_push.sh:50]: Currently, AudioQnA, SearchQnA, Translation and VisualQnA are not in the same folder structure, if we use the same steps to handle these workloads, we will meet with errors. Let me separate the workloads into two types: supported types and not supported types. After these workloads make changes, we can add them to the supported types.

File: .github/workflows/manifest-e2e.yaml (hunk 2 above)
Comment (daisy-ycguo) [reply .github/workflows/manifest-e2e.yaml]: resolved.

**Review by chensuyue — APPROVED** [review]

**Pull request merged**